react-components/src/architecture/base/concreteCommands/pointCloud/PointCloudFilterCommand.ts

```diff
@@ -193,30 +180,42 @@ export class PointClass {
 // PRIVATE FUNCTIONS
 // ==================================================
 
-function getFirstPointCloudWithClasses({
-  renderTarget
-}: {
-  renderTarget: RevealRenderTarget;
-}): PointCloud | undefined {
-  for (const pointCloud of renderTarget.getPointClouds()) {
-    const classes = pointCloud.getClasses();
-    if (classes === undefined || classes.length === 0) {
-      continue;
+function getFirstPointCloudWithClasses(root: RootDomainObject): PointCloudDomainObject | undefined {
+  for (const domainObject of root.getDescendantsByType(PointCloudDomainObject)) {
+    if (hasSomeClasses(domainObject.model)) {
+      return domainObject;
     }
-    return pointCloud;
   }
   return undefined;
 }
 
+function getCurrentDomainObject(
+  root: RootDomainObject,
+  uniqueId: UniqueId
+): PointCloudDomainObject | undefined {
+  const domainObject = root.getThisOrDescendantByUniqueId(uniqueId);
+  if (domainObject === undefined) {
+    return undefined;
+  }
+  return domainObject as PointCloudDomainObject;
```

Comment:
This is not type safe, can be updated as

```suggestion
  return domainObject instanceof PointCloudDomainObject ? domainObject : undefined
```